api/logs/views.py

```diff
@@ -0,0 +1,43 @@
+from modularodm import Q
+from rest_framework import generics, permissions as drf_permissions
+
+from website.models import Node, NodeLog
+
+from api.base.filters import ODMFilterMixin
+from api.logs.serializers import NodeLogSerializer
+from api.nodes.serializers import NodeSerializer
+
+class LogList(generics.ListAPIView, ODMFilterMixin):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+    serializer_class = NodeLogSerializer
+    ordering = ('-date', )  # default ordering
+
+    # overrides ODMFilterMixin
+    def get_default_odm_query(self):
+        allowed_node_ids = set(Node.find(Q('is_public', 'eq', True)).get_keys())
```

Comment:
@chrisseto are you sure about that? Just plucking pks loads all nodes?

If so I'm failing to see an alternative.